metaflow/runtime.py

```diff
@@ -238,8 +238,23 @@ def execute(self):
                                         'by the end of flow.')
 
     def _killall(self):
-        for worker in self._workers.values():
-            worker.kill()
+        # If we are here, all children have received a signal and are shutting down.
+        # We want to give them an opportunity to do so and then kill
+        live_workers = list(set(self._workers.values()))
```

Comment:
Why do you need this be a list? It seems you just need iteration, and the below suggested `filter`.

Reply:
sure, I can change it. It made more sense for it to be a list when I had a loop with add but now I can put just a set.